### ISSUE TITLE
Allow single-line if blocks with curly brackets

### ISSUE THREAD

**Issue by ndimiduk** [checkstyle/checkstyle#8881]
Allow single-line if blocks with curly brackets

```
$ javac CheckStyle8881.java 
$

$ cat checkstyle-8881.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
  "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
  "https://checkstyle.org/dtds/configuration_1_3.dtd" >
<module name="Checker">
  <module name="TreeWalker">
    <module name="NeedBraces" />
    <module name="LeftCurly">
      <property name="option" value="nlow" /> <!-- nlow -->
    </module>
    <module name="RightCurly"/>
    <module name="RightCurly">
      <property name="option" value="alone" />
      <property name="tokens" value="CLASS_DEF,METHOD_DEF,CTOR_DEF,STATIC_INIT,INTERFACE_DEF"/>
    </module>
  </module>
</module>

$ cat CheckStyle8881.java 
/**
 * There isn't a configuration combination that allows for a simple single-line
 * <pre>
 * if (condition) { statement; }
 * </pre>
 * While also requiring LeftCurly eol for method definitions.
 */
class CheckStyle8881 {

    void method0() { System.out.println("method"); } // "should have line break after. [LeftCurly]", "should be alone on a line. [RightCurly]"

    void method1() {
        if (System.out == null) { return; }
    }

    void method2() {
        if (System.out == null) { int i = 0; return; // "should have line break after. [LeftCurly]"
        } else {
            System.out.println();
            System.out.println();
        }
    }
}

# with LeftCurly=eol :
$ java -Duser.language=en -Duser.country=US -jar checkstyle-8.36.2-all.jar -c checkstyle-8881.xml CheckStyle8881.java 
Starting audit...
[ERROR] ~/tmp/CheckStyle8881.java:10:20: '{' at column 20 should have line break after. [LeftCurly]
[ERROR] ~/tmp/CheckStyle8881.java:10:52: '}' at column 52 should be alone on a line. [RightCurly]
[ERROR] ~/tmp/CheckStyle8881.java:13:33: '{' at column 33 should have line break after. [LeftCurly]
[ERROR] ~/tmp/CheckStyle8881.java:17:33: '{' at column 33 should have line break after. [LeftCurly]
Audit done.
Checkstyle ends with 4 errors.

# with LeftCurly=nlow
$ java -Duser.language=en -Duser.country=US -jar checkstyle-8.36.2-all.jar -c checkstyle-8881.xml CheckStyle8881.java 
Starting audit...
[ERROR] ~/tmp/CheckStyle8881.java:10:52: '}' at column 52 should be alone on a line. [RightCurly]
Audit done.
Checkstyle ends with 1 errors.
```
---------------

I seek a configuration that allows for simple single-line condition while also requiring While also requiring LeftCurly eol for method definitions.

--------------

# Earlier feature request content

**Is your feature request related to a problem? Please describe.**
I would like checkstyle (perhaps the `LeftCurly` check) to permit an if-block to accept a one-line then-block that is encased in curly brackets.

**Describe the solution you'd like**
This is an extension conceptually to the single-line if statement. I'm not sure how it would be described, maybe as an option to the `LeftCurly` check.

As an example,
```java
if (condition) { statement; }
```

The above solicits a complaint from the `LeftCurly` check, saying an EOL was expected after the `{` character.

**Additional context**
This style is useful for some situations where an early return is desired (if indeed you subscribe to that philosophy). For example, a [double-checked lock](https://en.wikipedia.org/wiki/Double-checked_locking#Usage_in_Java) with early exit instead of deep-nesting could be more concise using single-line conditional as described above.

**Comment by romani** [comment]
Please describe your issue as it it is a bug, as we demand in bug template. This will clarify all details.

You might want https://checkstyle.org/property_types.html#RightCurlyOption but I am not sure.

**Comment by ndimiduk** [comment]
> Please describe your issue as it it is a bug, as we demand in bug template.

Sorry, I saw it more as a feature request than bug. Let me try again. You want a new issue, or may I make edits to the above? Thanks @romani .

**Comment by romani** [comment]
Please edit this issue description

**Comment by ndimiduk** [comment]
Spent some time building the example as required by the bug template. It looks like `LeftCurly`'s `nlow` loosens up some of the requirements, but not in ways that I expected.

**Comment by romani** [comment]
Please update PR description with details to let me see what do you mean

**Comment by ndimiduk** [comment]
Thank you for your patience, @romani. I'm still waiting on legal approval for posting this code snippet. Sadly, I don't have an ETA.

**Comment by romani** [comment]
@ndimiduk ,

> legal approval for posting this code snippet.

you do not need this, and we do not need your exact secret code.
Please minimize your code to what is really required, obfuscate your code as much as required and share simple meaning less code snippet (all we need to just to be compilable code). You can see in other issues examples of code, non of them make any sense but can show a problem.

**Comment by ndimiduk** [comment]
I wish you were correct on this, but sadly, that's not how this company works. Anyway, I have approval. Let me make appropriate updates.

**Comment by saksham9656** [comment]
Hi i am interested in this issue. would you please guide me ?

**Comment by romani** [comment]
We need to agree on design before implementation, issue us not approved yet. We need to agree on design and way to implement it first.

**Comment by saksham9656** [comment]
hi, i want to contribute in checkstyle so would you please tell me more about this that what to do and what it is basically about.
I am a beginner so would you please guide me?

**Comment by mancze** [comment]
Wouldn't `RightCurlyOption` `same` or `alone_or_singleline` work for you?
https://checkstyle.sourceforge.io/property_types.html#RightCurlyOption

**Comment by ndimiduk** [comment]
I think this is done with with `LeftCurly` option `nlow`.
